### PR TITLE
Allow fastembed embedding model params configuration in parity with fastembed APIs

### PR DESCRIPTION
**Pull request by praveen-palanisamy**
This PR allows the user to configure the following parameters for the `fastembed` Embeddings via the `qdrant_client.set_model_params(...)` method:
```
max_length (int, optional): The maximum number of tokens.
cache_dir (str, optional): The path to the cache directory.
threads (int, optional): The number of threads single onnxruntime session can use.
```
in addition to the `embedding_model_name` parameter.  This brings parity with the init args of `fastembed`'s `Embedding` classes: `DefaultEmbedding`, `FlagEmbedding` and `JinaEmbedding`. This would allow a clean migration for existing `fastembed` users and switch to the simpler `qdrant_client.add(...)` API.
- [x] Added tests for the `qdrant_client.set_model_params(...)` method.

- [x] This fixes the issue described in #415 .

### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you installed `pre-commit` with `pip3 install pre-commit` and set up hooks with `pre-commit install`?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by DiTo97** [comment]
@praveen-palanisamy, when will it be merged?

**Comment by joein** [comment]
Hi @praveen-palanisamy , thanks for your contribution!

Hi, @DiTo97 I'll try to review it today and see whether we can merge it as is or not

**Comment by praveen-palanisamy** [comment]
@DiTo97 : Thanks for your interest and the ping. I am waiting for the review/approval before this can be merged. I'll follow-it up.

@joein : Do you have any feedback/review comments to finalize & merge this PR?

**Comment by joein** [comment]
Hi @praveen-palanisamy 

Yeap, I have some comments

Firstly, I'd like to avoid changing names of the existing API

Secondly, I would like to have `**kwargs` in the method's signature

Tbh, I am not even sure at the moment if we actually need to have `max_length`, `cache_dir` and `threads` as explicit params in the signature, or not. I need to get some context from the team. 

P.S. We use python3.10 to generate async client (there is a difference with parentheses starting from python3.11, CI will fail if you generate it with python != 3.10)

**Comment by joein** [comment]
Related https://github.com/qdrant/qdrant-client/pull/360

**Comment by praveen-palanisamy** [comment]
@joein Thanks for your comments.

> Firstly, I'd like to avoid changing names of the existing API
1. Okay. I proposed to change the name from `set_model` to `set_model_params` because it was more appropriate since this method is not only setting the model but also the associated config params. This new name also matched with the other existing (internal) method naming: `_get_model_params`. I can revert the name change if you prefer to keep it unchanged.

> Secondly, I would like to have **kwargs in the method's signature
2. Okay. I will update the signature. I am keeping the positional args as is for now with the `**kwargs` unused. Please see response below:

>Tbh, I am not even sure at the moment if we actually need to have max_length, cache_dir and threads as explicit params in the signature, or not. I need to get some context from the team.

3. Are you suggesting to just fold them all into `**kwargs`? They are kept as positional arguments in the same order as the underlying init args of fastembed's Embedding classes: `DefaultEmbedding`, `FlagEmbedding` and `JinaEmbedding`. Keeping this parity is preferred from a user point of view. I guess the `fastembed` API signature may change as it is early in its development cycle. If we want to avoid updating this signature frequently and bring back the parity at a later stage, it makes sense to me. Please confirm the change request after your sync with the team.

>P.S. We use python3.10 to generate async client (there is a difference with parentheses starting from python3.11, CI will fail if you generate it with python != 3.10)
4. Thanks for the tip! I just noticed that the CI runs [this check](https://github.com/qdrant/qdrant-client/blob/master/tests/async-client-consistency-check.sh) only with Python `3.10.x` while the test matrix includes Python versions `>=3.8 <=3.12`. [This check](https://github.com/qdrant/qdrant-client/blob/master/tests/async-client-consistency-check.sh) passes with Python 3.11 as well. Any particular reason why Python `3.10.x` alone was chosen to run that check/test?

Thanks for the pointer to # 360. That was not enough to allow me as a user to use `qdrant_client.add(...)` with the desired ONNX session option.

File: qdrant_client/qdrant_fastembed.py
Comment (joein) [inline qdrant_client/qdrant_fastembed.py:65]: **kwargs

File: qdrant_client/qdrant_fastembed.py
Comment (joein) [inline qdrant_client/qdrant_fastembed.py:111]: **kwargs

File: qdrant_client/qdrant_fastembed.py
Comment (joein) [inline qdrant_client/qdrant_fastembed.py:95]: **kwargs

**Comment by joein** [comment]
1. Yes, I prefer to avoid changes to the existing interface, if there is no necessity.
2-3. I think that we can have these parameters as keyword arguments, it is not hard to hide them inside `**kwargs` later if there is a need for this.
4. As I've told, python3.10 and python3.11 generates different code (3.10 adds parentheses, 3.11 does not). We can't have this check turned on for 3.10 and 3.11 simultaneously, since only one version can pass at the same moment. 3.10 was chosen due to the problems with gRPC generator.

**Comment by praveen-palanisamy** [comment]
@joein : Thanks for the comments and responses! I have resolved the remaining items and also regenerated the async APIs using Python 3.10.x.
Please let me know if any more changes are required before final CI check & merge.

**Comment by joein** [comment]
@praveen-palanisamy 

Fine by me, thank you for the contribution!

**Review by joein — APPROVED** [review]

**Pull request merged**
